FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -5158,16 +5158,19 @@ OrderedNode *OpDispatchBuilder::GetRelocatedPC(FEXCore::X86Tables::DecodedOp con
   return _EntrypointOffset(IR::SizeToOpSize(GPRSize), Op->PC + Op->InstSize + Offset - Entry);
 }
 
-OrderedNode *OpDispatchBuilder::LoadGPRRegister(uint32_t GPR, int8_t Size, uint8_t Offset) {
+OrderedNode *OpDispatchBuilder::LoadGPRRegister(uint32_t GPR, int8_t Size, uint8_t Offset, bool AllowUpperGarbage) {
   const uint8_t GPRSize = CTX->GetGPRSize();
   if (Size == -1) {
     Size = GPRSize;
   }
   OrderedNode *Reg = _LoadRegister(false, offsetof(FEXCore::Core::CPUState, gregs[GPR]), GPRClass, GPRFixedClass, GPRSize);
 
-  if (Size != GPRSize || Offset != 0) {
+  if ((!AllowUpperGarbage && (Size != GPRSize)) || Offset != 0) {
     // Extract the subregister if requested.
-    Reg = _Bfe(IR::SizeToOpSize(std::max<uint8_t>(4u, Size)), Size * 8, Offset, Reg);
+    if (AllowUpperGarbage)
+      Reg = _Lshr(IR::SizeToOpSize(std::max<uint8_t>(4u, Size)), Reg, _Constant(Offset));
```

Comment:
Maybe pull the

```cpp
IR::SizeToOpSize(std::max<uint8_t>(4u, Size))
```

bit into something like:

```cpp
const auto OpSize = IR::SizeToOpSize(std::max<uint8_t>(4u, Size));
```

So it doesn't need to be duplicated across both branches